frontend/src/components/transactions/Transactions.tsx

```diff
@@ -26,48 +28,67 @@ export default class extends React.PureComponent<Props, State> {
   };
 
   _transactionsApi: TransactionsApi | null;
+  timer: any;
 
   constructor(props: Props) {
     super(props);
 
     // TODO: Design ExplorerApi to handle server-side rendering gracefully.
     this._transactionsApi = null;
+    this.timer = 0;
   }
 
   componentDidMount() {
-    this.fetchTransactions();
+    this.regularFetchInfo();
   }
 
-  componentDidUpdate(prevProps: Props) {
-    if (this.props !== prevProps) {
-      this.fetchTransactions();
-    }
+  componentWillUnmount() {
+    clearTimeout(this.timer);
+    this.timer = null;
   }
 
-  render() {
-    const { transactions } = this.state;
-    if (transactions === null) {
-      return null;
+  regularFetchInfo = async () => {
+    await this.fetchTransactions();
+    if (this.timer !== null) {
+      this.timer = setTimeout(this.regularFetchInfo, 10000);
     }
-    return (
-      <TransactionsList
-        transactions={transactions}
-        reversed={this.props.reversed}
-      />
-    );
-  }
+  };
 
   fetchTransactions = async () => {
     if (this._transactionsApi === null) {
       this._transactionsApi = new TransactionsApi();
     }
-    const transactions = await this._transactionsApi.getTransactions({
-      signerId: this.props.accountId,
-      receiverId: this.props.accountId,
-      blockHash: this.props.blockHash,
-      tail: this.props.reversed,
-      limit: this.props.limit
-    });
+    const transactions = await new TransactionsApi()
+      .getLatestTransactionsInfo()
+      .catch(() => null);
     this.setState({ transactions });
   };
+
+  render() {
+    const { transactions } = this.state;
+    if (transactions === null) {
+      return null;
+    }
+    let reserved = true;
+    let actions = transactions.map(transaction =>
+      transaction.actions.map((action, actionIndex) => (
+        <ActionRow
+          key={transaction.hash + actionIndex}
+          action={action}
+          transaction={transaction}
+          viewMode="compact"
+        />
+      ))
+    );
+
+    if (reserved) {
+      actions.reverse();
+    }
+
+    return (
+      <FlipMove duration={1000} staggerDurationBy={0}>
+        {actions}
+      </FlipMove>
+    );
+  }
```

Comment:
Let's try reverting this inlining of the TransactionsList -> ActionsList components as it gets really mixed and duplicated. I believe, the problem is just that we need to use class-based components for TransactionsList and ActionsList as FlipMove requires `ref`, which is not available for function components.

If we succeed here, we should fix `DashboardTransactions.jsx` as well.